modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -284,16 +311,20 @@ private DefaultPollContext(SourceCallback<T, A> sourceCallback, Serializable cur
       this.updatedWatermark = updatedWatermark;
       this.currentPollItems = 0;
       this.minimumRejectedByLimitPassingWatermark = null;
+      this.timestamp = ZonedDateTime.now();
+    }
+
+    public String getPollId() {
+      return componentLocation.getLocation() + " | " + timestamp;
```

Comment:
We should discuss if we want the id make this id to have this information included. A random UID is also a possibility

Reply:
We talked about this in the spec stage, but we can reopen the discussion. As Composer will display this ID we thought it would be best for it to be as user friendly as possible. What are the advantages of a random UID that you think of?

Reply:
I think that since the composer user does not directly create the mule app, maybe the names that composer uses for flows may not be the most descriptive. This will also add a /source to the flow name. 

I am not against this option, since the flow name will give useful information, just bringing this to the table.

Reply:
@ndinu: the feature needs to be generic and not tailored to how composer may or may not name flows
@SebaElizalde: The full component location doesn't make sense as these are sources and they'll always be at the same coordinate. What you should use is the container flow name. Also, `@` is a more clear separator than `|`